src/cargo/core/compiler/build_runner/compilation_files.rs

```diff
@@ -922,3 +924,14 @@ fn use_extra_filename(bcx: &BuildContext<'_, '_>, unit: &Unit) -> bool {
     }
     true
 }
+
+/// Returns whether or not this unit should use a hash in the pkg_dir to make it unique.
+fn use_pkg_dir(unit: &Unit) -> bool {
+    if unit.mode.is_doc_test() || unit.mode.is_doc() {
+        // Doc tests do not have metadata.
```

Comment:
This seems fine for now until we want to store rustdoc artifacts separately.

A future idea, with `-Zrustdoc-mergeable-info`, we could keep the each rustdoc artifacts separately. And later we do a merge that 

* Uplift everything at the end.
* Run `rustdoc --merge=finalize` then


A downside of it: we'll need to copy a lot of files. Probably not worthy